src/ui/public/kuery/suggestions/value.js

```diff
@@ -1,16 +1,33 @@
+import 'isomorphic-fetch';
 import { flatten, memoize } from 'lodash';
 import chrome from '../../chrome';
 import { escapeQuotes } from './escape_kuery';
 
 const baseUrl = chrome.addBasePath('/api/kibana/suggestions/values');
 const type = 'value';
 
-export function getSuggestionsProvider({ $http, config, indexPatterns }) {
-  const allFields = flatten(indexPatterns.map(indexPattern => indexPattern.fields.raw));
-  const requestSuggestions = memoize((query, field) => {
-    const queryParams = { query, field: field.name };
-    return $http.post(`${baseUrl}/${field.indexPattern.title}`, queryParams);
-  }, resolver);
+const requestSuggestions = memoize(async (query, field) => {
+  const response = await fetch(`${baseUrl}/${field.indexPatternTitle}`, {
+    method: 'POST',
+    body: JSON.stringify({ query, field: field.name }),
+    headers: new Headers({
+      'Content-Type': 'application/json',
+      'kbn-xsrf': true
+    }),
```

Comment:
Using fetch seems fine to me but we should probably check with some other folks on the team who have done more reactification to see if they've established any best practices or have built any existing services. It would be nice if we didn't have to enumerate the headers here, in case the required headers ever change (for example we're missing the kbn-version header here which isn't strictly required, but I think pretty much every other kibana request includes it).

@Stacey-Gammon you've done a lot of de-angularization, do you know if anything like this exists already?

Reply:
We have this open ticket for deciding on a library to use: https://github.com/elastic/dev/issues/849  @nreese - do you know if the wrapper around fetch service was built for the Home stuff?

Reply:
the home application did not build a re-usable wrapper around fetch.

Reply:
It sounds like a decision was made to use fetch, but I couldn't find any http client. If you don't mind, I'll start this effort with a super minimalistic http client based on fetch. To eat my own dog food I'll make sure it also works for APM.

Reply:
Added a very simple http client called kfetch (Kibana wrapper around fetch - naming suggestions very welcome).
https://github.com/sqren/kibana/blob/f93d6befb3e1e7d4751d78ee82622d2309bd089a/src/ui/public/kfetch/index.js

I intentionally didn't add any helper methods like `.post`, `.asJson` etc. People can implement that on top of if they want. All this does is adding the kibana specific headers that is required by everyone, to a normal fetch request.

Reply:
@sqren you might want to chat with @spalger. I thought he was looking into a wrapper around fetch.

Reply:
@nreese cool, thanks. 
@spalger I didn't see anything in master. Is it work-in-progress? If you haven't started yet we can use my client as a starting point.

Reply:
@sqren I haven't started on it yet: https://github.com/elastic/kibana/issues/18744

Reply:
I love the little `kfetch` helper BTW, looks perfect